django_rest_api/user/serializers.py

```diff
@@ -24,52 +46,14 @@ def validate(self, data):
             raise serializers.ValidationError("Must include 'username' and 'password'.")
         
         user = authenticate(username=username, password=password)
-        if user:
-            data['user'] = user
-        else:
+        if not user:
             raise serializers.ValidationError("Invalid credentials.")
-        
+
+        data['user'] = user
         return data
 
 
     def create(self, validated_data):
```

Comment:
Having this `create` method fetch an object rather than create one is not the best (you're also doing a second DB call for the user - `authenticate` function above does one more). You don't need this method.
Generally, what you can do with login flow is use the `authenticate` Django method inside the view rather than the serializer. See my comment below.